### PR TITLE
Add inner template test case for APIVersion-Should-Be-Recent

### PR DESCRIPTION
**Pull request by galiacheng**
Calling an API in the inner template, test case `apiVersions-Should-Be-Recent` constructs the resource type by adding prefix "Microsoft.Resources/deployments", then getting resource type like "Microsoft.Resources/deployments/Microsoft.Network/dnsZones", which causes the test failed. This case should pass.

Error example:

```text
[-] apiVersions Should Be Recent (5359 ms)
        The apiVersion 2020-07-01 was not found for the resource type: M
icrosoft.Resources/deployments/Microsoft.Network/networkSecurityGroups
        The apiVersion 2020-07-01 was not found for the resource typ
e: Microsoft.Resources/deployments/Microsoft.Network/virtualNetworks
        The apiVersion 2020-07-01 was not found for the resource typ
e: Microsoft.Resources/deployments/Microsoft.Network/publicIPAddresses
        The apiVersion 2020-07-01 was not found for the resource typ
e: Microsoft.Resources/deployments/Microsoft.Network/applicationGateways
        Api versions must be the latest or under 2 years old (730 da
ys) - API version 2018-05-01 of Microsoft.Resources/deployments/Microsoft.Netwo
rk/dnsZones is 1290 days old
        Valid Api Versions:
        2021-01-01
        2021-01-01
        2020-10-01
        2020-06-01
        The apiVersion 2021-06-01-preview was not found for the resource
 type: Microsoft.Resources/deployments/Microsoft.KeyVault/vaults
        The apiVersion 2021-06-01-preview was not found for the reso
urce type: Microsoft.Resources/deployments/Microsoft.KeyVault/vaults
        The apiVersion 2021-06-01-preview was not found for the reso
urce type: Microsoft.Resources/deployments/Microsoft.KeyVault/vaults/secrets
        The apiVersion 2021-06-01-preview was not found for the reso
urce type: Microsoft.Resources/deployments/Microsoft.KeyVault/vaults/secrets
        The apiVersion 2021-06-01-preview was not found for the reso
urce type: Microsoft.Resources/deployments/Microsoft.KeyVault/vaults
        The apiVersion 2021-06-01-preview was not found for the reso
urce type: Microsoft.Resources/deployments/Microsoft.KeyVault/vaults/secrets
        The apiVersion 2021-02-01 was not found for the resource typ
e: Microsoft.Storage/storageAccounts/fileServices/shares
        The apiVersion 2020-08-01 was not found for the resource typ
e: Microsoft.Resources/deployments/Microsoft.OperationalInsights/workspaces
        The apiVersion 2021-02-01 was not found for the resource typ
e: Microsoft.Resources/deployments/Microsoft.ContainerService/managedClusters
        The apiVersion 2021-02-01 was not found for the resource typ
e: Microsoft.Resources/deployments/Microsoft.ContainerService/managedClusters
        The apiVersion 2020-11-01-preview was not found for the reso
urce type: Microsoft.Resources/deployments/Microsoft.ContainerRegistry/registri
es
        The apiVersion 2021-02-01 was not found for the resource typ
e: Microsoft.Resources/deployments/Microsoft.Storage/storageAccounts
        The apiVersion 2021-02-01 was not found for the resource typ
e: Microsoft.Resources/deployments/Microsoft.Storage/storageAccounts/fileServic
es/shares
        The apiVersion 2021-02-01 was not found for the resource typ
e: Microsoft.Storage/storageAccounts/fileServices/shares
        The apiVersion 2020-08-01 was not found for the resource typ
e: Microsoft.Resources/deployments/Microsoft.OperationalInsights/workspaces
        The apiVersion 2021-02-01 was not found for the resource typ
e: Microsoft.Resources/deployments/Microsoft.ContainerService/managedClusters
        The apiVersion 2021-02-01 was not found for the resource typ
e: Microsoft.Resources/deployments/Microsoft.ContainerService/managedClusters
        The apiVersion 2020-11-01-preview was not found for the reso
urce type: Microsoft.Resources/deployments/Microsoft.ContainerRegistry/registri
es
        The apiVersion 2021-02-01 was not found for the resource typ
e: Microsoft.Resources/deployments/Microsoft.Storage/storageAccounts
        The apiVersion 2021-02-01 was not found for the resource typ
e: Microsoft.Resources/deployments/Microsoft.Storage/storageAccounts/fileServic
es/shares
```

### CODE REVIEW

File: arm-ttk/testcases/deploymentTemplate/apiVersions-Should-Be-Recent.test.ps1 [hunk 1]
```diff
@@ -86,8 +86,19 @@ foreach ($av in $allApiVersions) {
             # by walking backwards over the parent resources
             # (since the topmost resource will be the last item in the list)
             for ($i = $av.ParentObject.Count - 1; $i -ge 0; $i--) {
-                if (-not $av.ParentObject[$i].type) { continue }
-                $av.ParentObject[$i].type
+                $parent=$av.ParentObject[$i]
```
Comment (bmoore-msft) [inline arm-ttk/testcases/deploymentTemplate/apiVersions-Should-Be-Recent.test.ps1:89]: format the code, e.g. SHIFT+ALT+F in VS Code

File: arm-ttk/testcases/deploymentTemplate/apiVersions-Should-Be-Recent.test.ps1 [hunk 2]
```diff
@@ -86,8 +86,19 @@ foreach ($av in $allApiVersions) {
...
+                if (-not $parent.type) { continue }
+
+                # if parent resource type is Microsoft.Resources/deployments, and this is an inner template, skip
+                $expEvaOptions= $parent.properties.expressionEvaluationOptions
```
Comment (bmoore-msft) [inline arm-ttk/testcases/deploymentTemplate/apiVersions-Should-Be-Recent.test.ps1:93]: Should probably name the var ```expEvalOptions``` (add an L)

File: arm-ttk/testcases/deploymentTemplate/apiVersions-Should-Be-Recent.test.ps1 [hunk 3]
```diff
@@ -86,8 +86,19 @@ foreach ($av in $allApiVersions) {
...
+                if ($parent.type -eq "Microsoft.Resources/deployments" -and $expEvaOptions) {
+                    $scope=$expEvaOptions.scope
+                    if ($scope -eq "inner") {
+                        continue
+                    }
+                }
+
```
Comment (bmoore-msft) [inline arm-ttk/testcases/deploymentTemplate/apiVersions-Should-Be-Recent.test.ps1:100]: remove whitespace

File: unit-tests/apiversions-should-be-recent/Pass/Api-In-Inner-Template.json [hunk 4]
```diff
@@ -0,0 +1,65 @@
+{
```
Comment (bmoore-msft) [inline unit-tests/apiversions-should-be-recent/Pass/Api-In-Inner-Template.json:1]: Let's add a fail test case as well

File: arm-ttk/testcases/deploymentTemplate/apiVersions-Should-Be-Recent.test.ps1 [hunk 5]
```diff
@@ -86,8 +86,19 @@ foreach ($av in $allApiVersions) {
...
-                $av.ParentObject[$i].type
+                $parent=$av.ParentObject[$i]
+                if (-not $parent.type) { continue }
+
+                # if parent resource type is Microsoft.Resources/deployments, and this is an inner template, skip
+                $expEvaOptions= $parent.properties.expressionEvaluationOptions
+                if ($parent.type -eq "Microsoft.Resources/deployments" -and $expEvaOptions) {
+                    $scope=$expEvaOptions.scope
+                    if ($scope -eq "inner") {
+                        continue
```
Comment (bmoore-msft) [inline arm-ttk/testcases/deploymentTemplate/apiVersions-Should-Be-Recent.test.ps1:97]: Not sure I follow here though - we don't want to skip the test, we just don't want to add the parent of ```Microsoft.Resources/deployment``` right?

File: arm-ttk/cache/AllAzureResources.cache.json [hunk 6]
```diff
@@ -29184,6 +29184,11 @@
       "aliases": null,
       "apiProfiles": null,
       "apiVersions": [
+        "2021-04-01",
```
Comment (bmoore-msft) [inline arm-ttk/cache/AllAzureResources.cache.json:29187]: this file is autogenerated so these changes will be overwritten - so let's just remove them.

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: arm-ttk/testcases/deploymentTemplate/apiVersions-Should-Be-Recent.test.ps1 (hunk 5 above)
Comment (galiacheng) [reply arm-ttk/testcases/deploymentTemplate/apiVersions-Should-Be-Recent.test.ps1:97]: Yes, I will enhance the comment

File: arm-ttk/testcases/deploymentTemplate/apiVersions-Should-Be-Recent.test.ps1 (hunk 1 above)
Comment (galiacheng) [reply arm-ttk/testcases/deploymentTemplate/apiVersions-Should-Be-Recent.test.ps1]: Fixed in https://github.com/Azure/arm-ttk/pull/540/commits/fb722d374d7c6942ede76ffc0a088a83ad8e8691

File: arm-ttk/testcases/deploymentTemplate/apiVersions-Should-Be-Recent.test.ps1 (hunk 2 above)
Comment (galiacheng) [reply arm-ttk/testcases/deploymentTemplate/apiVersions-Should-Be-Recent.test.ps1]: Fixed in https://github.com/Azure/arm-ttk/pull/540/commits/fb722d374d7c6942ede76ffc0a088a83ad8e8691

File: arm-ttk/testcases/deploymentTemplate/apiVersions-Should-Be-Recent.test.ps1 (hunk 3 above)
Comment (galiacheng) [reply arm-ttk/testcases/deploymentTemplate/apiVersions-Should-Be-Recent.test.ps1]: Fixed in https://github.com/Azure/arm-ttk/pull/540/commits/fb722d374d7c6942ede76ffc0a088a83ad8e8691

File: unit-tests/apiversions-should-be-recent/Pass/Api-In-Inner-Template.json (hunk 4 above)
Comment (galiacheng) [reply unit-tests/apiversions-should-be-recent/Pass/Api-In-Inner-Template.json:1]: Fix in https://github.com/Azure/arm-ttk/pull/540/commits/fb722d374d7c6942ede76ffc0a088a83ad8e8691

File: arm-ttk/testcases/deploymentTemplate/apiVersions-Should-Be-Recent.test.ps1 (hunk 5 above)
Comment (galiacheng) [reply arm-ttk/testcases/deploymentTemplate/apiVersions-Should-Be-Recent.test.ps1:98]: Fixed in https://github.com/Azure/arm-ttk/pull/540/commits/fb722d374d7c6942ede76ffc0a088a83ad8e8691

File: arm-ttk/cache/AllAzureResources.cache.json (hunk 6 above)
Comment (galiacheng) [reply arm-ttk/cache/AllAzureResources.cache.json]: Done

**Pull request merged**